ckan/logic/validators.py

```diff
@@ -49,6 +49,27 @@ def owner_org_validator(key, data, errors, context):
             raise Invalid(_('An organization must be provided'))
         return
 
+    if (authz.check_config_permission('allow_dataset_collaborators')
+            and not authz.check_config_permission('allow_collaborators_to_change_owner_org')):
+
+        package = context.get('package')
+        if package and user and not user.sysadmin:
+            try:
+                user_packages = logic.get_action(
+                    'package_member_list_for_user')(
+                            {'ignore_auth': True}, {'id': user.id})
```

Comment:
same comment as above, if you'd like to avoid db calls all over the place maybe add a helper just for checking the collaborator status for a single package id + user id.